presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -262,9 +266,13 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
             PartitioningHandle partitioning,
             List<TableScanNode> tableScans,
             Map<PlanNodeId, SplitSource> splitSources,
-            Optional<Integer> numberOfShufflePartitions)
+            Optional<Integer> numberOfShufflePartitions,
+            Optional<NativeExecutionNode> nativeExecution)
     {
         ListMultimap<Integer, SerializedPrestoSparkTaskSource> taskSourcesMap = ArrayListMultimap.create();
+        boolean nativeExecutionEnabled = isNativeExecutionEnabled(session);
+        checkArgument(nativeExecutionEnabled == nativeExecution.isPresent(),
+                "Can't find the NativeExecutionNode in when native engine execution is enabled");
```

Comment:
Message has a typo, and I would change it to: "NativeExecutionNode not present in plan, but native engine execution is enabled"